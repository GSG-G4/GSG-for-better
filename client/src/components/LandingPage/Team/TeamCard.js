import React from 'react';
import PropTypes from 'prop-types';

import {
  CardWrapper,
  ImageContainer,
  CardImage,
  UserName,
  UserSpecialty,
} from './StyledCompnents';

import CardImageBackground from './team_card.png';

const TeamCard = ({ name, profileImage, specialty }) => (
  <CardWrapper>
    <ImageContainer img={CardImageBackground}>
      <CardImage src={profileImage} />
    </ImageContainer>
    <UserName>{name}</UserName>
    <UserSpecialty>{specialty}</UserSpecialty>
  </CardWrapper>
);

export default TeamCard;

TeamCard.propTypes = {
  name: PropTypes.string.isRequired,
  profileImage: PropTypes.string.isRequired,
  specialty: PropTypes.string.isRequired,
};
