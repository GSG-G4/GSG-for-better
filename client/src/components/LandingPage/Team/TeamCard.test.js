import React from 'react';
import renderer from 'react-test-renderer';

import TeamCard from './TeamCard';

test('Testing for TeamCard component', () => {
  const component = renderer.create(
    <TeamCard
      name="Ramy A Shurafa"
      profileImage="https://avatars3.githubusercontent.com/u/34629478?s=460&v=4"
      specialty="Full-Stack Web Developer"
    />
  );
  expect(component).toMatchSnapshot();
});
