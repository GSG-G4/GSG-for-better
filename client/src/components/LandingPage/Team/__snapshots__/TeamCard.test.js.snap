// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for TeamCard component 1`] = `
<div
  className="sc-bdVaJa mdupv"
>
  <div
    className="sc-bwzfXH hqawGO"
  >
    <img
      className="sc-htpNat jvkVJd"
      src="https://avatars3.githubusercontent.com/u/34629478?s=460&v=4"
    />
  </div>
  <h1
    className="sc-bxivhb jcTKvB"
  >
    Ramy A Shurafa
  </h1>
  <h2
    className="sc-ifAKCX hhiBCA"
  >
    Full-Stack Web Developer
  </h2>
</div>
`;
